zcash_primitives/src/legacy/keys.rs

```diff
@@ -405,6 +413,27 @@ impl private::SealedChangeLevelKey for InternalIvk {
 
 impl IncomingViewingKey for InternalIvk {}
 
+/// An incoming viewing key at the "ephemeral" path
+/// `m/44'/<coin_type>'/<account>'/2`.
+///
+/// This allows derivation of ephemeral addresses for use within the wallet.
+#[derive(Clone, Debug)]
+pub struct EphemeralIvk(ExtendedPublicKey<PublicKey>);
+
+impl private::SealedChangeLevelKey for EphemeralIvk {
+    const SCOPE: TransparentKeyScope = TransparentKeyScope(2);
+
+    fn extended_pubkey(&self) -> &ExtendedPublicKey<PublicKey> {
+        &self.0
+    }
+
+    fn from_extended_pubkey(key: ExtendedPublicKey<PublicKey>) -> Self {
+        EphemeralIvk(key)
+    }
+}
+
+impl IncomingViewingKey for EphemeralIvk {}
```

Comment:
Aside: `IncomingViewingKey::derive_address` is deprecated, but I don't see how to implement its functionality at all from outside the crate (since it requires access to `zcash_primitives::legacy::keys::private::SealedChangeLevelKey`). What's the intended replacement?